src/pyhf/__init__.py

```diff
@@ -4,10 +4,8 @@
 from .exceptions import InvalidBackend
 from . import events
 
-tensorlib = tensor.numpy_backend()
-default_backend = tensorlib
-optimizer = optimize.scipy_optimizer()
-default_optimizer = optimizer
+tensorlib = None
+optimizer = None
```

Comment:
Why do we need to split these up into here and the section below?

Reply:
cyclic import. the default optimizer (scipy) needs the `get_backend` function call which depends on those variables being global, but they can't be set until after `get_backend` is defined.